examples/src/tower/client.rs

```diff
@@ -57,13 +58,9 @@ mod service {
         }
 
         fn call(&mut self, req: Request<BoxBody>) -> Self::Future {
-            let mut channel = self.inner.clone();
+            // Do extra async work here...
 
-            Box::pin(async move {
-                // Do extra async work here...
-
-                channel.call(req).await.map_err(Into::into)
-            })
+            Box::pin(self.inner.call(req).err_into::<Self::Error>())
```

Comment:
```suggestion
        let clone = self.inner.clone();
        let mut inner = std::mem::replace(&mut self.inner, clone);
        Box::pin(async move {
            // Do extra async work here...
            inner.call(req).await.map_err::<Self::Error>()
        })
```

This was we still make it clear to users where to add additional async code.